src/webui/www/private/scripts/client.js

```diff
@@ -353,6 +411,66 @@ window.addEvent('load', function() {
         }
     };
 
+    const updateTagList = function() {
+        const tagFilterList = $('tagFilterList');
+        if (!tagFilterList)
+            return;
+
+        while (tagFilterList.firstChild) {
+            tagFilterList.removeChild(tagFilterList.firstChild);
+        }
+
+        const createLink = function(hash, text, count) {
+            const html = '<a href="#" onclick="setTagFilter(' + hash + ');return false;">'
+                + '<img src="images/qbt-theme/inode-directory.svg"/>'
+                + escapeHtml(text) + ' (' + count + ')' + '</a>';
+            const el = new Element('li', {
+                id: hash,
+                html: html
+            });
+            tagsFilterContextMenu.addTarget(el);
+            return el;
+        };
+
+        const torrentsCount = torrentsTable.getRowIds().length;
+        let untagged = 0;
+        for (let key in torrentsTable.rows) {
+            if (torrentsTable.rows.hasOwnProperty(key)) {
+                const row = torrentsTable.rows[key];
+                if (row['full_data'].tags.length === 0)
+                    untagged += 1;
+            }
+        }
+        tagFilterList.appendChild(createLink(TAGS_ALL, 'QBT_TR(All)QBT_TR[CONTEXT=TagFilterModel]', torrentsCount));
+        tagFilterList.appendChild(createLink(TAGS_UNTAGGED, 'QBT_TR(Untagged)QBT_TR[CONTEXT=TagFilterModel]', untagged));
+
+        const sortedTags = [];
+        for (let key in tagList)
+            sortedTags.push(tagList[key].name);
+        sortedTags.sort();
+
+        for (let key in sortedTags) {
+            if (sortedTags.hasOwnProperty(key)) {
```

Comment:
there are a few `hasOwnProperty()` around, I presume you are sure that they are really required?